pkg/virtualrouter/enqueue_requests_for_mesh_events.go

```diff
@@ -0,0 +1,71 @@
+package virtualrouter
+
+import (
+	"context"
+	appmesh "github.com/aws/aws-app-mesh-controller-for-k8s/apis/appmesh/v1beta2"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/pkg/k8s"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/pkg/mesh"
+	"github.com/go-logr/logr"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+)
+
+func NewEnqueueRequestsForMeshEvents(k8sClient client.Client, log logr.Logger) *enqueueRequestsForMeshEvents {
+	return &enqueueRequestsForMeshEvents{
+		k8sClient: k8sClient,
+		log:       log,
+	}
+}
+
+var _ handler.EventHandler = (*enqueueRequestsForMeshEvents)(nil)
+
+type enqueueRequestsForMeshEvents struct {
+	k8sClient client.Client
+	log       logr.Logger
+}
+
+// Create is called in response to an create event
+func (h *enqueueRequestsForMeshEvents) Create(e event.CreateEvent, queue workqueue.RateLimitingInterface) {
```

Comment:
So create and delete calls are no-ops because that is handled outside of this reconcile loop?

Reply:
because virtualNode only cares whether mesh is active :D